frontend/packages/ceph-storage-plugin/src/components/dashboards/odf-system-dashboard.tsx

```diff
@@ -28,28 +30,32 @@ const ODFSystemDashboard: React.FC<ODFSystemDashboardPageProps> = ({
   const { t } = useTranslation();
   const isObjectServiceAvailable = useFlag(MCG_FLAG);
   const isCephAvailable = useFlag(CEPH_FLAG);
+  const isIndependent = useFlag(OCS_INDEPENDENT_FLAG);
   const { systemName } = rest.match.params;
   const dashboardTab = !isCephAvailable && isObjectServiceAvailable ? OBJECT : BLOCK_FILE;
   const defaultDashboard = React.useRef(dashboardTab);
-
-  const pages: Page[] = [
+  const [pages, setPages] = React.useState<Page[]>([
     {
       path: 'overview/:dashboard',
       href: `overview/${defaultDashboard.current}`,
       name: t('ceph-storage-plugin~Overview'),
       component: OCSOverview,
     },
-  ];
+  ]);
 
   React.useEffect(() => {
-    if (isCephAvailable) {
-      pages.push({
-        href: referenceForModel(CephBlockPoolModel),
-        name: t('ceph-storage-plugin~BlockPools'),
-        component: () => <BlockPoolListPage namespace={CEPH_STORAGE_NAMESPACE} />,
-      });
+    const isBlockPoolAdded = pages.find((page) => page.href === blockPoolRef);
+    if (isCephAvailable && !isObjectServiceAvailable && !isIndependent && !isBlockPoolAdded) {
+      setPages([
```

Comment:
Why do we need a state here ? And not just adding conditionals for mcg and external mode to push block pool page.
Can you explain ?

```suggestion
```

Reply:
This change came in when I rebased this PR on top of Bipul's PR that fixed a related issue, so that we can test this. Here's [Bipul's comment](https://github.com/openshift/console/pull/10537/files#r754834684) where he explains this change.

Reply:
Removed that commit from this PR since it's already merged.